packages/backend/src/workers/provider/LlamaCppPython.ts

```diff
@@ -254,8 +275,29 @@ export class LlamaCppPython extends InferenceProvider {
       case VMType.LIBKRUN_LABEL:
         return gpu ? llamacpp.vulkan : llamacpp.default;
       // no GPU support
+      case VMType.UNKNOWN:
+        return this.isNvidiaCDIConfigured(gpu) ? llamacpp.cuda : llamacpp.default;
       default:
         return llamacpp.default;
     }
   }
+
+  protected isNvidiaCDIConfigured(gpu?: IGPUInfo): boolean {
```

Comment:
You don't need to take the gpu as argument, this function name is not GPU specific, it is to check nvidia CDI, so when called, juste check for known location

Reply:
@axel7083 it does this check

if (gpu?.vendor !== GPUVendor.NVIDIA) return false;

which also guards againts there being no gpu or gpu accelerating not being enabled.  If we don't do the check in the function we'd have to do it in addition to calling isNvidiaCDIConfigured. I think we could either:

1) change the name of the function
2) do the additional check before calling the function.

I think its less code if we do 1). @axel7083 which would you prefer ?

Reply:
In terms of a new function name maybe something like `useGpuOnLinux`.  The function is called in two places so I think doing the check inside the shared function does make sense.

Reply:
Let's merge this for now, I re-read after your comment, and this is an acceptable solution